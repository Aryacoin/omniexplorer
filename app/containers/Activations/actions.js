/*
 *
 * Activations actions
 *
 */

import {
  LOAD_ACTIVATIONS,
  LOAD_ACTIVATIONS_SUCCESS,
} from './constants';

/**
 * Load activations, this action starts the request saga
 *
 * @return {object} An action object with a type of LOAD_ACTIVATIONS
 */
export function loadActivations() {
  return {
    type: LOAD_ACTIVATIONS,
  };
}
