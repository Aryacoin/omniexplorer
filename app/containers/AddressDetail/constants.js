/*
 *
 * AddressDetail constants
 *
 */

export const DEFAULT_ACTION = 'app/AddressDetail/DEFAULT_ACTION';
