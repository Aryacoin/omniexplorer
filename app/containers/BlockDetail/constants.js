/*
 *
 * BlockDetail constants
 *
 */

export const LOAD_BLOCK = 'omniexplorer/App/LOAD_BLOCK';
export const LOAD_BLOCK_SUCCESS = 'omniexplorer/App/LOAD_BLOCK_SUCCESS';
