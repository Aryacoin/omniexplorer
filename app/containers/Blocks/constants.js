/*
 *
 * Blocks constants
 *
 */

export const LOAD_BLOCKS = 'omniexplorer/App/LOAD_BLOCKS';
export const LOAD_BLOCKS_SUCCESS = 'omniexplorer/App/LOAD_BLOCKS_SUCCESS';
export const LOAD_BLOCKS_ERROR = 'omniexplorer/App/LOAD_BLOCKS_ERROR';
