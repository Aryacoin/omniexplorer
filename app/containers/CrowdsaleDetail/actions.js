import {
  LOAD_CROWDSALE_TRANSACTIONS,
  LOAD_CROWDSALE_TRANSACTIONS_SUCCESS,
  LOAD_CROWDSALE_TRANSACTIONS_ERROR,
  SET_CROWDSALES_TRANSACTIONS_PAGE,
} from './constants';

/**
 * Log an error if the async call fails
 * @param {object} error - the error thrown.
 */
export const errorCrowdsaleTransactionsFetch = error => ({
  type: LOAD_CROWDSALE_TRANSACTIONS_ERROR,
  error,
});

// Start the fetch, toggle is `isFetching` value
export const startCrowdsaleTransactionsFetch = crowdsaleId => ({
  type: LOAD_CROWDSALE_TRANSACTIONS,
  id: crowdsaleId,
});

/**
 * Resolve the fetch with the returned data
 * @param {object} payload - the data returned from the fetch
 */
export const updateCrowdsaleTransactionsFetch = (
  transactions,
  pages,
  total,
  page,
) => ({
  type: LOAD_CROWDSALE_TRANSACTIONS_SUCCESS,
  transactions,
  pages,
  total,
  page,
});

/**
 * Dispatched when the page change
 *
 * @param  {number} page The page number
 *
 * @return {object} An action object with the page
 */
export function setPage(page) {
  return {
    type: SET_CROWDSALES_TRANSACTIONS_PAGE,
    page,
  };
}
