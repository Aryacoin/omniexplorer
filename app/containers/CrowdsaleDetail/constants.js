/*
 *
 * CrowdsaleDetail constants
 *
 */

export const DEFAULT_ACTION = 'app/CrowdsaleDetail/DEFAULT_ACTION';
