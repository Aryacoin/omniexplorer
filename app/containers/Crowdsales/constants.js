/*
 *
 * Search constants
 *
 */

export const LOAD_CROWDSALES = 'omniexplorer/App/LOAD_CROWDSALES';
export const LOAD_CROWDSALES_SUCCESS = 'omniexplorer/App/LOAD_CROWDSALES_SUCCESS';
export const LOAD_CROWDSALES_ERROR = 'omniexplorer/App/LOAD_CROWDSALES_ERROR';
export const LOAD_CROWDSALES_ECOSYSTEM_PROD = 1;
export const LOAD_CROWDSALES_ECOSYSTEM_TEST = 2;

