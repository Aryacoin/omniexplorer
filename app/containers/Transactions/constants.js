/*
 *
 * Transactions constants
 *
 */

export const LOAD_TRANSACTIONS = 'omniexplorer/App/LOAD_TRANSACTIONS';
export const LOAD_TRANSACTIONS_SUCCESS =
  'omniexplorer/App/LOAD_TRANSACTIONS_SUCCESS';
export const SET_PAGE = 'omniexplorer/App/SET_PAGE';
export const SET_TRANSACTION_TYPE = 'omniexplorer/App/SET_TRANSACTION_TYPE';
export const LOAD_UNCONFIRMED = 'omniexplorer/App/LOAD_UNCONFIRMED';
export const LOAD_CLASSAB_TXS = 'omniexplorer/App/LOAD_CLASSAB_TXS';
export const LOAD_CLASSAB_TXS_BLOCKCHAIN_INFO_SUCCESS = 'omniexplorer/App/LOAD_CLASSAB_TXS_BLOCKCHAIN_INFO_SUCCESS';
export const LOAD_CLASSAB_TXS_SUCCESS =
  'omniexplorer/App/LOAD_CLASSAB_TXS_SUCCESS';
export const CONFIRMATIONS = 100;
export const TRANSACTION_TYPE = {
  CONFIRMED: 'confirmed',
  UNCONFIRMED: 'unconfirmed',
  CLASSABTX: 'classABTx',
};
