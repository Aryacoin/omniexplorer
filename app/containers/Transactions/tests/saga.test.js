/**
 * Tests for Transactions sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { LOAD_TRANSACTIONS } from '../constants';
import { transactionsLoaded, transactionsLoadingError } from '../actions';

import root, { getTransactions } from '../saga';

const txid = 'dbf8b73aa9149ae3e8a96e85c64c48d8061d65c026b16c899e77bb6a607bd45x';

/* eslint-disable redux-saga/yield-effects */
describe('getTransaction Saga', () => {
  let getTransactionGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getTransactionGenerator = getTransactions();

    const selectDescriptor = getTransactionGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = getTransactionGenerator.next(txid).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the transactionsLoaded action if it requests the data successfully', () => {
    const response = {
      pages: 143938,
      transactions: [
        {
          amount: '65067.68000000',
          blocktime: 1522349840,
          confirmations: 0,
          divisible: true,
          fee: '0.00001283',
          ismine: false,
          propertyid: 31,
          propertyname: 'TetherUS',
          referenceaddress: '1DkvAif2AKfRrd5GZya5U1QdDAG8hCoucx',
          sendingaddress: '17ScKNXo4cL8DyfWfcCWu1uJySQuJm7iKx',
          txid: 'dbf8b73aa9149ae3e8a96e85c64c48d8061d65c026b16c899e77bb6a607bd45x',
          type: 'Simple Send',
          type_int: 0,
          version: 0,
        }],
    };

    const putDescriptor = getTransactionGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(transactionsLoaded(response, txid)));
  });

  it('should call the transactionsLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getTransactionGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(transactionsLoadingError(response)));
  });
});

describe('githubDataSaga Saga', () => {
  const rootSaga = root();

  it('should start task to watch for LOAD_TRANSACTIONS action', () => {
    const takeLatestDescriptor = rootSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_TRANSACTIONS, getTransactions));
  });
});
