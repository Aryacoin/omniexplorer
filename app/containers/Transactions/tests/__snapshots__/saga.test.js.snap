// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTransaction Saga should call the transactionsLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getTransaction Saga should call the transactionsLoadingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "currentPage": 0,
      "pages": undefined,
      "transactions": undefined,
      "type": "omniexplorer/App/LOAD_TRANSACTIONS_SUCCESS",
    },
    "channel": null,
  },
}
`;

exports[`getTransaction Saga should dispatch the transactionsLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`getTransaction Saga should dispatch the transactionsLoaded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "currentPage": 0,
      "pages": undefined,
      "transactions": undefined,
      "type": "omniexplorer/App/LOAD_TRANSACTIONS_SUCCESS",
    },
    "channel": null,
  },
}
`;
